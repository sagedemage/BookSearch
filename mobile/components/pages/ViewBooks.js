import { useEffect, useState } from 'react';
import axios from 'axios';
import { View, FlatList, Text } from 'react-native';

// 10.0.2.2:3000

export default function ViewBooks() {
    const [books, setBooks] = useState([]);

    useEffect(() => {
        axios.get('http://10.0.2.2:5000/search?q=c+programming+language')
            .then(function(response) {
                setBooks(response.data.books)
            })
            .catch(function(error) {
                console.error(error);
            })
    }, []);

    const _renderItem = ({ item }) => (
        <View style={{ flexDirection: 'row'}}>
            {/* Only 3 or 4 items a time */}
            <View style={{width: 100}}>
                <Text style={{textAlign: 'center'}}>{item.title}</Text>
            </View>
            <View style={{width: 100}}>
                <Text style={{textAlign: 'center'}}>{item.author_name}</Text>
            </View>
            <View style={{width: 100}}>
                <Text style={{textAlign: 'center'}}>{item.first_publish_year}</Text>
            </View>
            <View style={{width: 100}}>
                <Text style={{textAlign: 'center'}}>{item.number_of_pages_median}</Text>
            </View>
        </View>
    );

    return (
        <View style={{flex: 1, justifyContent: 'center', alignItems: 'center', marginTop: '10%'}}>
            <View style={{ flexDirection: 'row'}}>
                <View style={{width: 100, backgroundColor: 'black'}}>
                    <Text style={{textAlign: 'center', color: 'white'}}>Title</Text>
                </View>
                <View style={{width: 100, backgroundColor: 'black'}}>
                    <Text style={{textAlign: 'center', color: 'white'}}>Author Name</Text>
                </View>
                <View style={{width: 100, backgroundColor: 'black'}}>
                    <Text style={{textAlign: 'center', color: 'white'}}>First Publish Year</Text>
                </View>
                <View style={{width: 100, backgroundColor: 'black'}}>
                    <Text style={{textAlign: 'center', color: 'white'}}>Number of Pages Median</Text>
                </View>
            </View>
            <FlatList
                data={books}
                renderItem={_renderItem} 
                />               
        </View>
    );
}