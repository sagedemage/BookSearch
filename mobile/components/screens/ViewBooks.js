import { React, useState } from 'react';
import axios from 'axios';
import { View, FlatList, Text, TextInput, Button, ActivityIndicator, Image } from 'react-native';
import { styles, tableCell } from '../../styles';
import { SERVER_URL } from '../../config';

/**
 * View Books Screen Component
 * @returns {React.ReactHTMLElement} Returns the HTML element
 */
export default function ViewBooks() {
  const [books, setBooks] = useState([]);
  const [text, setText] = useState('');
  const [isLoading, setLoading] = useState(false);

  const viewBooks = () => {
    setLoading(true);
    axios.get(SERVER_URL + '/api/search?q=' + text,
    ).then((response) => {
      setBooks(response.data.books);
    }).catch((error) => {
      console.error(error);
    }).finally(() => {
      setLoading(false);
    });
  };

  const _renderItem = ({ item }) => (
    <View style={styles.row}>
      <View style={tableCell(85, 'lightgray')}>
        <Image
          style={styles.book_cover}
          source={{ uri: item.image_url }}
          alt='book cover image' />
      </View>
      <View style={tableCell(300)}>
        <Text style={{ fontSize: 16 }}>{item.title}</Text>
        <Text style={{ color: 'gray', fontSize: 12 }}>
          by {item.author_name}
        </Text>
        <Text style={{ fontSize: 12 }}>
          Publish Year: {item.first_publish_year}
        </Text>
        <Text style={{ fontSize: 12 }}>
          Number of Pages: {item.number_of_pages_median}
        </Text>
      </View>
    </View>
  );

  return (
    <View style={styles.container}>
      <View style={styles.search_box}>
        <Button title='Search' onPress={() => viewBooks()}></Button>
        <TextInput
          style={styles.text_input}
          placeholder="Search Books"
          onChangeText={(newText) => setText(newText)}
          defaultValue={text}
        ></TextInput>
      </View>
      <View style={styles.table}>
        {isLoading ? (
          <ActivityIndicator />
        ) : (
          <FlatList
            data={books}
            renderItem={_renderItem}
          />
        )}
      </View>
    </View>
  );
}

