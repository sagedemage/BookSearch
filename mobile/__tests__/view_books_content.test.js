import * as React from 'react';
import renderer from 'react-test-renderer';
import ViewBooks from '../components/screens/ViewBooks';

test('check the view books screen content matches with the snapshot', () => {
    /* Content for ViewBooks.js  */
    const tree = renderer
      .create(<ViewBooks />)
      .toJSON();
  
    expect(tree).toMatchSnapshot();
  });