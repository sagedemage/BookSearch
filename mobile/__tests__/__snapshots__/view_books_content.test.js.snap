// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the view books screen content matches with the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginTop": 10,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Search
        </Text>
      </View>
    </View>
    <TextInput
      defaultValue=""
      onChangeText={[Function]}
      placeholder="Search Books"
      style={
        {
          "borderWidth": 2,
          "height": 35,
          "paddingHorizontal": 10,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": "5%",
      }
    }
  >
    <View
      style={
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "black",
            "padding": 5,
            "width": 95,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          Title
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "black",
            "padding": 5,
            "width": 95,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          Author Name
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "black",
            "padding": 5,
            "width": 95,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          First Publish Year
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "black",
            "padding": 5,
            "width": 95,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "textAlign": "center",
            }
          }
        >
          Number of Pages Median
        </Text>
      </View>
    </View>
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
